shared/entities/cost-inputs/blue-carbon-project-planning.entity.ts

```diff
@@ -35,6 +42,18 @@ export class BlueCarbonProjectPlanning extends BaseEntity {
   @Column("decimal", { name: "input_3_cost_per_project" })
   input3: number;
 
-  @OneToOne(() => BaseData, (baseData) => baseData.blueCarbonProjectPlanning)
-  baseData: BaseData;
+  @Column("decimal", { name: "blue_carbon" })
+  blueCarbon: number;
+
+  @BeforeInsert()
+  @BeforeUpdate()
```

Comment:
Nice! I think this work on a transactional level, meaning it will be triggered for each transaction over this entity, no matter if ita a sinle insert/update or bulk

I guess this works for us